sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/GenericInheritanceShouldNotBeRecursive.CSharp9.cs

```diff
@@ -0,0 +1,11 @@
+﻿record CA<T> { }
+record CB<T> { }
+record CC<T1, T2> { }
+record C0<T> : CA<C0<T>> { } // Compliant
+record C1<T> : CA<CB<C1<T>>> { } // Compliant
+record C2<T> : CA<C2<CB<T>>> { } // Compliant - FN
+record C3<T> : CA<C3<C3<T>>> { } // Compliant - FN
+record C4<T> : CA<C4<CA<T>>> { } // Compliant - FN
+record C5<T> : CC<C5<CA<T>>, CB<T>> { } // Compliant - FN
+record C6<T> : CC<C5<CA<T>>, CB<T>> { } // Compliant
+record C7<T> : CA<CA<CA<CA<CA<CA<CA<CA<CA<CA<CA<CA<C7<CB<T>>>>>>>>>>>>>> { } // Compliant - FN
```

Comment:
```suggestion
record C2<T> : CA<C2<CB<T>>> { } // FN
record C3<T> : CA<C3<C3<T>>> { } // FN
record C4<T> : CA<C4<CA<T>>> { } // FN
record C5<T> : CC<C5<CA<T>>, CB<T>> { } // FN
record C6<T> : CC<C5<CA<T>>, CB<T>> { } // Compliant
record C7<T> : CA<CA<CA<CA<CA<CA<CA<CA<CA<CA<CA<CA<C7<CB<T>>>>>>>>>>>>>> { } // FN
```

Reply:
as a generic note: `Compliant - FN` was long time used as a convention

Reply:
Let's stop the leak :)